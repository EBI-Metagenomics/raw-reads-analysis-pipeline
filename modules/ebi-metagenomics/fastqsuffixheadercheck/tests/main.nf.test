
nextflow_process {

    name "Test Process FASTQSUFFIXHEADERCHECK"
    script "../main.nf"
    process "FASTQSUFFIXHEADERCHECK"

    tag "modules"
    tag "modules_nfcore"
    tag "fastqsuffixheadercheck"

    test("fastqsuffixheadercheck - pass - SE") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true ], // meta map
                    file("${moduleDir}/tests/fixtures/ERR4334351.fastq.gz", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("fastqsuffixheadercheck - pass - PE") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file("${moduleDir}/tests/fixtures/ERR4674038_1.fastq.gz", checkIfExists: true),
                      file("${moduleDir}/tests/fixtures/ERR4674038_2.fastq.gz", checkIfExists: true)
                    ]
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("fastqsuffixheadercheck - fail - PE") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file("${moduleDir}/tests/fixtures/ERR4674038witherror_1.fastq.gz", checkIfExists: true),
                      file("${moduleDir}/tests/fixtures/ERR4674038witherror_2.fastq.gz", checkIfExists: true)
                    ]
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("fastqsuffixheadercheck - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/ERR4334351.fastq.gz", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
