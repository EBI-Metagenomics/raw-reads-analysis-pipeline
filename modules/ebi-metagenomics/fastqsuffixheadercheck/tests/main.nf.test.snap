{
    "fastqsuffixheadercheck - fail - PE": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,735222eb9c22d4d81b312a9682d55679"
                    ]
                ],
                "1": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,735222eb9c22d4d81b312a9682d55679"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T12:00:40.71854211"
    },
    "fastqsuffixheadercheck - pass - PE": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T12:00:33.408673717"
    },
    "fastqsuffixheadercheck - pass - SE": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T12:00:25.910535012"
    },
    "fastqsuffixheadercheck - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cd9316e4086b766d70f49b7a69557c26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T12:00:48.417924407"
    }
}